docs/source/python/integration/python_java.rst

```diff
@@ -0,0 +1,429 @@
+    [INFO] Copying flatbuffers-java-1.12.0.jar to /experiments/java2py/dependencies/flatbuffers-java-1.12.0.jar
+    [INFO] Copying arrow-memory-8.0.0-SNAPSHOT.pom to /experiments/java2py/dependencies/arrow-memory-8.0.0-SNAPSHOT.pom
+    [INFO] Copying netty-buffer-4.1.72.Final.jar to /experiments/java2py/dependencies/netty-buffer-4.1.72.Final.jar
+    [INFO] Copying jackson-databind-2.11.4.jar to /experiments/java2py/dependencies/jackson-databind-2.11.4.jar
+    [INFO] Copying commons-codec-1.10.jar to /experiments/java2py/dependencies/commons-codec-1.10.jar
+    [INFO] ------------------------------------------------------------------------
+    [INFO] BUILD SUCCESS
+    [INFO] ------------------------------------------------------------------------
+
+Once our package and all its depdendencies are available, 
+we can invoke it from ``fillten_pyarrowjvm.py`` script that will
+import the ``FillTen`` class and print out the result of invoking ``FillTen.createArray`` 
+
+.. code-block:: python
+
+    import jpype
+    import jpype.imports
+    from jpype.types import *
+
+    # Start a JVM making available all dependencies we collected
+    # and our class from target/FillTen-1.jar
+    jpype.startJVM(classpath=["./dependencies/*", "./target/*"])
+
+    FillTen = JClass('FillTen')
+
+    array = FillTen.createArray()
+    print("ARRAY", type(array), array)
+
+    # Convert the proxied ValueVector to an actual pyarrow array
+    import pyarrow.jvm
+    pyarray = pyarrow.jvm.array(array)
+    print("ARRAY", type(pyarray), pyarray)
+    del pyarray
+
+Running the python script will lead to two lines getting printed:
+
+.. code-block::
+
+    ARRAY <java class 'org.apache.arrow.vector.BigIntVector'> [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+    ARRAY <class 'pyarrow.lib.Int64Array'> [
+        1,
+        2,
+        3,
+        4,
+        5,
+        6,
+        7,
+        8,
+        9,
+        10
+    ]
+
+The first line is the raw result of invoking the ``FillTen.createArray`` method.
+The resulting object is a proxy to the actual Java object, so it's not really a pyarrow
+Array, it will lack most of its capabilities and methods. 
+That's why we subsequently use ``pyarrow.jvm.array`` to convert it to an actual
+``pyarrow`` array. That allows us to treat it like any other ``pyarrow`` array.
+The result is the second line in the output where the array is correctly reported
+as being of type ``pyarrow.lib.Int64Array`` and is printed using the ``pyarrow`` style.
+
+Java to Python communication using the C Data Interface
+-------------------------------------------------------
+
+The C-Data interface is a protocol implemented in Arrow to exchange data within different
+environments without the cost of marshaling and copying data.
+
+This allows to expose data coming from Python or Java to functions that are implemented
+in the other language.
+
```

Comment:
Add or validate if cffi module is already installed: `pip install cffi`

Reply:
:+1: